src/Nethermind/Nethermind.Core/Bloom.cs

```diff
@@ -108,86 +129,107 @@ public override bool Equals(object? obj)
 
         public override int GetHashCode()
         {
-            return Bytes.GetSimplifiedHashCode();
+            return ((ReadOnlySpan<byte>)Bytes).GetSimplifiedHashCode();
         }
 
         public void Add(LogEntry[] logEntries, Bloom? blockBloom = null)
         {
-            for (int entryIndex = 0; entryIndex < logEntries.Length; entryIndex++)
+            for (int i = 0; i < logEntries.Length; i++)
             {
-                LogEntry logEntry = logEntries[entryIndex];
+                LogEntry logEntry = logEntries[i];
                 byte[] addressBytes = logEntry.LoggersAddress.Bytes;
+
                 Set(addressBytes, blockBloom);
-                for (int topicIndex = 0; topicIndex < logEntry.Topics.Length; topicIndex++)
+
+                Keccak[] topics = logEntry.Topics;
+                if (topics.Length > 0)
                 {
-                    Keccak topic = logEntry.Topics[topicIndex];
-                    Set(topic.Bytes, blockBloom);
+                    AddTopics(blockBloom, topics);
                 }
             }
         }
 
+        private void AddTopics(Bloom? blockBloom, Keccak[] topics)
+        {
+            for (int i = 0; i < topics.Length; i++)
+            {
+                Keccak topic = topics[i];
+                Set(topic.Bytes, blockBloom);
+            }
+        }
+
         public void Accumulate(Bloom bloom)
         {
-            Bytes.AsSpan().Or(bloom.Bytes);
+            ref BloomData data = ref bloom._data;
+            for (int i = 0; i < Bloom.Vector256Length; i++)
+            {
+                Vector256<byte> v1 = Vector256.LoadUnsafe(ref Unsafe.AddByteOffset(ref Unsafe.As<BloomData, byte>(ref _data), i * Bloom.Vector256Size));
+                Vector256<byte> v2 = Vector256.LoadUnsafe(ref Unsafe.AddByteOffset(ref Unsafe.As<BloomData, byte>(ref data), i * Bloom.Vector256Size));
+                Vector256.StoreUnsafe(Vector256.BitwiseOr(v1, v2), ref Unsafe.AddByteOffset(ref Unsafe.As<BloomData, byte>(ref _data), i * Bloom.Vector256Size));
+            }
```

Comment:
This already was using Vector256 underneath.

Reply:
Yeah but variable length with via span loading and only for x64 `Vector256.BitwiseOr` also works for Arm. Ends up with a ton more instructions